ruma-events/tests/any_enums.rs

```diff
@@ -0,0 +1,278 @@
+use std::convert::TryFrom;
+
+use matches::assert_matches;
+use ruma_identifiers::RoomAliasId;
+use serde_json::{from_value as from_json_value, json};
+
+use ruma_events::{
+    room::{
+        aliases::AliasesEventContent,
+        message::{MessageEventContent, TextMessageEventContent},
+        power_levels::PowerLevelsEventContent,
+    },
+    AnyEvent, AnyMessageEventContent, AnyRoomEvent, AnyRoomEventStub, AnyStateEventContent,
+    EventJson, MessageEvent, MessageEventStub, StateEvent, StateEventStub,
+};
+
+#[test]
+fn power_event_stub_deserialization() {
+    let json_data = json!({
+        "content": {
+            "ban": 50,
+            "events": {
+                "m.room.avatar": 50,
+                "m.room.canonical_alias": 50,
+                "m.room.history_visibility": 100,
+                "m.room.name": 50,
+                "m.room.power_levels": 100
+            },
+            "events_default": 0,
+            "invite": 0,
+            "kick": 50,
+            "redact": 50,
+            "state_default": 50,
+            "users": {
+                "@example:localhost": 100
+            },
+            "users_default": 0
+        },
+        "event_id": "$15139375512JaHAW:localhost",
+        "origin_server_ts": 45,
+        "sender": "@example:localhost",
+        "state_key": "",
+        "type": "m.room.power_levels",
+        "unsigned": {
+            "age": 45
+        }
+    });
+
+    assert_matches!(
+        from_json_value::<EventJson<AnyRoomEventStub>>(json_data)
+            .unwrap()
+            .deserialize()
+            .unwrap(),
+        AnyRoomEventStub::State(
+            StateEventStub {
```

Comment:
No need to round-trip through `EventJson` here either if we just `unwrap()` anyways. Same applies to the other tests.

```suggestion
        from_json_value::<AnyRoomEventStub>(json_data),
        Ok(AnyRoomEventStub::State(StateEventStub {
```